xchange-coingi/src/main/java/org/knowm/xchange/coingi/service/CoingiTradeService.java

```diff
@@ -0,0 +1,177 @@
+import org.knowm.xchange.service.trade.params.CancelOrderByIdParams;
+import org.knowm.xchange.service.trade.params.CancelOrderParams;
+import org.knowm.xchange.service.trade.params.TradeHistoryParamPaging;
+import org.knowm.xchange.service.trade.params.TradeHistoryParams;
+import org.knowm.xchange.service.trade.params.orders.DefaultOpenOrdersParamCurrencyPair;
+import org.knowm.xchange.service.trade.params.orders.OpenOrdersParams;
+
+public class CoingiTradeService extends CoingiTradeServiceRaw implements TradeService {
+  public CoingiTradeService(Exchange exchange) {
+    super(exchange);
+  }
+
+  @Override
+  public OpenOrders getOpenOrders() throws IOException {
+    return getOpenOrders(createOpenOrdersParams());
+  }
+
+  @Override
+  public OpenOrders getOpenOrders(OpenOrdersParams params) throws IOException {
+    CoingiGetOrderHistoryRequest orderHistoryRequest = new CoingiGetOrderHistoryRequest();
+    orderHistoryRequest.setStatus(0);
+    orderHistoryRequest.setPageNumber(1);
+    orderHistoryRequest.setPageSize(50);
+    try {
+      CoingiOrdersList list = getCoingiOrderHistory(orderHistoryRequest);
+      return CoingiAdapters.adaptOpenOrders(list);
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+  }
+
+  @Override
+  public String placeMarketOrder(MarketOrder order) throws IOException {
+    throw new NotYetImplementedForExchangeException();
+  }
+
+  @Override
+  public String placeStopOrder(org.knowm.xchange.dto.trade.StopOrder stopOrder)
+      throws java.io.IOException {
+    throw new NotYetImplementedForExchangeException();
+  }
+
+  @Override
+  public String placeLimitOrder(LimitOrder order) throws IOException {
+    CoingiPlaceLimitOrderRequest request =
+        new CoingiPlaceLimitOrderRequest()
+            .setCurrencyPair(CoingiAdapters.adaptCurrency(order.getCurrencyPair()))
+            .setOrderType(order.getType().equals(BID) ? 0 : 1)
+            .setPrice(order.getLimitPrice())
+            .setVolume(order.getRemainingAmount());
+
+    String response;
+    try {
+      response = placeCoingiLimitOrder(request).getResult();
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+
+    return response;
+  }
+
+  @Override
+  public boolean cancelOrder(String orderId) throws IOException {
+    // if it doesn't return an error and CoingiOrder is returned, then the cancellation is
+    // successful
+    CoingiCancelOrderRequest request = new CoingiCancelOrderRequest();
+    request.setOrderId(orderId);
+    CoingiOrder order;
+    try {
+      order = cancelCoingiOrder(request);
+    } catch (CoingiException e) {
+      throw CoingiErrorAdapter.adapt(e);
+    }
+
+    return order != null;
+  }
+
+  @Override
+  public boolean cancelOrder(CancelOrderParams orderParams) throws IOException {
+    if (orderParams instanceof CancelOrderByIdParams) {
+      return cancelOrder(((CancelOrderByIdParams) orderParams).getOrderId());
```

Comment:
shouldn't this be wrapped in the try / catch CoingiException ?

Reply:
done, thanks for spotting this